### ISSUE TITLE
MySQL 8: Change default auth on bootup via docker command

### ISSUE THREAD

**Issue by fideloper** [tighten/takeout#53]
MySQL 8 requires (still, I believe!) the use of the `--default-authentication-plugin=mysql_native_password` flag to work with Laravel (and really just stay inline with user expectations).

> @josecanhelp - let me know if I'm wrong on that, I still use 5.7 for the most part!

One way of doing that would be to edit the default service templates to add a `$cmd` variable, for example, MySQL's:

```php
protected $dockerRunTemplate = '-p "$port":3306 \
        -e MYSQL_ROOT_PASSWORD="$root_password" \
        -v "$volume":/var/lib/mysql \
        "$organization"/"$image_name":"$tag" "$cmd"';
```

Then we'd have to handle allowing a custom `$cmd` to be used when starting a container. 

> For MySQL, the `ENTRYPOINT` is (for all intents and purposes) `mysqld`. The Docker `CMD` can just be flags to pass the `mysqld` command, like so:
> 
> ```bash
> docker run -d mysql:8 --default-authentication-plugin=mysql_native_password
> ```

Adding a `$cmd` to the templates would allow users to add that flag if they wanted.

This addition may mean another default prompt for services, but perhaps it could be a hidden option for advanced usage?

Another idea in this case is to code up a ✨special case conditional✨😅 where MySQL 8+ gets this flag appended to `$cmd`.

**Comment by morloderex** [comment]
@fideloper This is not entirely true. 

As far is i know if you are using PDO_MYSQL with PHP above 7.4.4. The new authentication plugin should not be any issue. I have fallowed along with PHP internals as I myself have had issues with this. 

The reason it was not supported in prior versions was due to the implementation been depending on the hash extension which is why the php internal choose to bundle that extension per default in 7.4.0.

What i'm trying to say here is that maybe we should detect the version and add this conditionally?
